### PR TITLE
Allow transfer of ownership of DbConnection from application to DbContext

### LINKED ISSUE #24199: Allow transfer of ownership of DbConnection from application to DbContext
The defaults:

- When EF Core creates a DbConnection from a connection string, then EF Core owns that connection and will dispose it when the context is disposed.
- When the application creates a DbConnection and passes to a DbContext, then the application still owns the context and is responsible for disposing it.

The proposal here is to allow an application to pass ownership of the DbConnection to DbContext so that DbContext will handle disposing it. This existing in EF6, and we just saw a customer scenario where this would be very useful in EF Core.

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #24199

### CODE REVIEW

**Comment by roji** [comment]
https://github.com/npgsql/efcore.pg/issues/2611 for the PG side

File: src/EFCore.SqlServer/Extensions/SqlServerDbContextOptionsBuilderExtensions.cs [hunk 1]
```diff
@@ -75,6 +75,30 @@ public static class SqlServerDbContextOptionsExtensions
...
+    ///     See <see href="https://aka.ms/efcore-docs-dbcontext-options">Using DbContextOptions</see>, and
+    ///     <see href="https://aka.ms/efcore-docs-sqlserver">Accessing SQL Server and SQL Azure databases with EF Core</see>
+    ///     for more information and examples.
+    /// </remarks>
+    /// <param name="optionsBuilder">The builder being used to configure the context.</param>
+    /// <param name="connection">
+    ///     An existing <see cref="DbConnection" /> to be used to connect to the database. If the connection is
+    ///     in the open state then EF will not open or close the connection. If the connection is in the closed
+    ///     state then EF will open and close the connection as needed. The caller owns the connection and is
+    ///     responsible for itd disposal.
```
Comment (AndriySvyryd) [inline src/EFCore.SqlServer/Extensions/SqlServerDbContextOptionsBuilderExtensions.cs:92]:
```suggestion
    ///     responsible for its disposal.
```

File: src/EFCore.SqlServer/Extensions/SqlServerDbContextOptionsBuilderExtensions.cs [hunk 2]
```diff
@@ -170,7 +200,8 @@ public static class SqlServerDbContextOptionsExtensions
     /// <param name="connection">
     ///     An existing <see cref="DbConnection" /> to be used to connect to the database. If the connection is
     ///     in the open state then EF will not open or close the connection. If the connection is in the closed
-    ///     state then EF will open and close the connection as needed.
+    ///     state then EF will open and close the connection as needed. The caller owns the connection and is
+    ///     responsible for itd disposal.
```
Comment (AndriySvyryd) [inline src/EFCore.SqlServer/Extensions/SqlServerDbContextOptionsBuilderExtensions.cs:204]:
```suggestion
    ///     responsible for its disposal.
```

File: src/EFCore.Sqlite.Core/Extensions/SqliteDbContextOptionsBuilderExtensions.cs [hunk 3]
```diff
@@ -161,7 +189,8 @@ public static class SqliteDbContextOptionsBuilderExtensions
     /// <param name="connection">
     ///     An existing <see cref="DbConnection" /> to be used to connect to the database. If the connection is
     ///     in the open state then EF will not open or close the connection. If the connection is in the closed
-    ///     state then EF will open and close the connection as needed.
+    ///     state then EF will open and close the connection as needed. The caller owns the connection and is
+    ///     responsible for itd disposal.
```
Comment (AndriySvyryd) [inline src/EFCore.Sqlite.Core/Extensions/SqliteDbContextOptionsBuilderExtensions.cs:193]:
```suggestion
    ///     responsible for its disposal.
```

**Review by AndriySvyryd — APPROVED** [review]

**Pull request merged**